fem/pgridfunc.cpp

```diff
@@ -225,23 +228,27 @@ void ParGridFunction::ExchangeFaceNbrData()
    MPI_Request *recv_requests = requests + num_face_nbrs;
    MPI_Status  *statuses = new MPI_Status[num_face_nbrs];
 
-   const double *h_data = this->HostRead();
-   for (int i = 0; i < send_data.Size(); i++)
+   auto d_data = this->Read();
```

Comment:
I think this implementation can be improved to handle better the case when gpu is enabled but `this->data` is on host when this method is called. In that case, it will be better to do the buffer copies on host avoiding the copy of the whole data to the gpu.

There's no need to change the implementation now, I just wanted to point this out for future considerations.